packages/core/src/lib/modal.svelte

```diff
@@ -26,81 +26,84 @@ Creates a modal overlay.
 
 <script lang="ts">
 import cx from 'classnames';
-import { createEventDispatcher, onMount } from 'svelte';
+import { createEventDispatcher } from 'svelte';
 import IconButton from './button/icon-button.svelte';
+import { clickOutside } from '$lib';
 
 /** Whether the modal is open. */
 export let open = false;
+
+$: if (typeof document !== 'undefined') {
+  document.body.classList.toggle('overflow-hidden', open);
+}
+
 /** The variant of the modal. */
 export let variant: 'small' | '' = '';
 
 const dispatch = createEventDispatcher<{
   /** Fires when the modal is requested to close. */
-  close: boolean;
+  close: void;
 }>();
 
 const handleCloseModal = () => {
-  dispatch('close', true);
-};
-
-const handleBackgroundClick = (event: MouseEvent) => {
-  if (event.target === event.currentTarget) {
-    handleCloseModal();
+  if (!open) {
+    return;
   }
+  dispatch('close');
 };
 
-const handleBackgroundKey = (event: KeyboardEvent) => {
-  if (event.key === 'Enter' || event.key === 'Space') {
+const handleEscapePress = (event: KeyboardEvent) => {
+  if (event.key === 'Escape') {
+    event.preventDefault();
     handleCloseModal();
   }
 };
-
-onMount(() => {
-  document.body.classList.toggle('overflow-hidden', open);
-});
 </script>
 
-<div
-  class={cx(
-    'fixed left-0 top-0 z-50 flex h-full w-full items-center justify-center bg-black bg-opacity-40',
-    {
-      invisible: !open,
-    }
-  )}
-  role="button"
-  tabindex="0"
-  on:click={handleBackgroundClick}
-  on:keydown={handleBackgroundKey}
->
+<svelte:window on:keydown={handleEscapePress} />
+
+{#if open}
   <div
-    class={cx('relative max-w-lg border border-medium bg-white p-6 shadow-sm', {
-      'max-h-[600px] min-h-[400px] w-[480px]': variant === '',
-      'max-h-[320px] w-[400px]': variant === 'small',
-    })}
+    class="fixed left-0 top-0 z-50 flex h-full w-full items-center justify-center bg-black bg-opacity-40"
+    role="dialog"
+    tabindex="-1"
```

Comment:
oops, this may have been my fault, i was reading the guidelines here https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/tabindex

do we still need to provide tabindex to prevent tabbing to the modal if it's not open? or maybe the aria role is enough to accommodate that?

Reply:
Thank you!  Updated accordingly, my test for this isn't cooperating, do you happen to have any suggestions?

```
  it('should focus on heading element on mount', () => {
    render(Modal, { isOpen });
    const heading = document.querySelector('#modal-heading');
    const elementInFocus = document.activeElement;
    expect(elementInFocus).toEqual(heading);
  });
```